core/mock_test.go

```diff
@@ -364,6 +364,17 @@ func (m *mockCluster) runSequence(height uint64) {
 	}
 }
 
+// runSequenceUntilHeight runs multiple sequence iterations until the desired height is reached
+func (m *mockCluster) runSequenceUntilHeight(desiredHeight uint64) {
```

Comment:
Just curious, is this a correct assumption that we wait on every step of the loop?

Reply:
Yes, for this specific case I want all of the nodes to finish the sequence correctly. I've removed this helper, and instead migrated it inside the test itself